programs/ssl/ssl_client2.c

```diff
@@ -2417,6 +2417,19 @@ int main( int argc, char *argv[] )
     mbedtls_printf( " %d bytes written in %d fragments\n\n%s\n",
                     written, frags, (char *) buf );
 
+    /* Make sure all data is flushed. */
+    while( 1 )
+    {
+        ret = mbedtls_ssl_flush( &ssl );
+        if( ret != MBEDTLS_ERR_SSL_WANT_WRITE )
```

Comment:
This program supports renegotiation so other errors are possible. Likewise for `ssl_server2`.

Reply:
We don't check for this in the previous version of the code, so I suggest we leave it for 3.X to improve the example programs in this regard.

Reply:
We do: the loop around `mbedtls_ssl_write` allows `MBEDTLS_ERR_SSL_CRYPTO_IN_PROGRESS` and `MBEDTLS_ERR_SSL_WANT_READ` as well as `MBEDTLS_ERR_SSL_WANT_WRITE`.